ehr/src/org/labkey/ehr/table/DefaultEHRCustomizer.java

```diff
@@ -824,7 +827,12 @@ public void renderGridCellContents(RenderContext ctx, Writer out) throws IOExcep
                         Date date = (Date) ctx.get("date");
                         Object id = ctx.get(ID_COL);
 
-                        out.write("<span style=\"white-space:nowrap\"><a href=\"javascript:void(0);\" onclick=\"EHR.window.ClinicalHistoryWindow.showClinicalHistory('" + objectid + "', '" + id + "', '" + date + "', this);\">[Show Hx]</a></span>");
+                        out.write("<span style=\"white-space:nowrap\"><a class=\"labkey-text-link anm-history\" data-objectid=\"" + PageFlowUtil.filter(objectid) + "\" data-id=\"" + PageFlowUtil.filter(id) + "\" data-date=\"" + PageFlowUtil.filter(date) + "\">[Show Hx]</a></span>");
```

Comment:
@ankurjuneja: I really thing you should just: 

1) no longer include data-objectid and data-date at all
2) on line 833 just pass 'null' for those arguments. 

There is no real downside, no change to code outside of what you're already touching here, you fix a potential bug should anyone every use 'date' for anything, and you end up writing less text to the page.